doc/plutus/explanations/plutus-tools-component-descriptions.rst

```diff
@@ -0,0 +1,321 @@
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+| Dependencies | cardano-api, plutus-ledger-api, plutus-ledger,                |
+|              | plutus-ledger-constraints, plutus-chain-index-core,           |
+|              | plutus-contract                                               |
++--------------+---------------------------------------------------------------+
+| State        | Maintained and will be extended. Unstable interface.          | 
++--------------+---------------------------------------------------------------+
+
+9. Plutus contract state machine
+-----------------------------------------------------------
+
+`Plutus contract state machine <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-contract>`_ 
+is a library that provides a high-level model for writing a Plutus application based 
+on the State Machine formalism. Even though it is inefficient, it can be used to 
+model a Plutus application and to test the actual implementation.
+
++--------------+---------------------------------------------------------------+
+| Plutus contract state machine                                                |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+| Dependencies | plutus-contract                                               |
++--------------+---------------------------------------------------------------+
+| State        | We maintain it, but it is not production ready. It is         |
+|              | inefficient---it creates Plutus applications that have high   |
+|              | Plutus script sizes; it is not scalable.                      | 
++--------------+---------------------------------------------------------------+
+
+10. Plutus contract testing model
+------------------------------------------
+
+`Plutus contract testing model <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-contract>`_ 
+is used for testing Plutus contracts with *contract models*, using the framework 
+provided by :hsmod:`Plutus.Contract.Test.ContractModel`. This framework generates
+and runs tests on the Plutus emulator, where each test may involve a number of 
+emulated wallets, each running a collection of Plutus contracts, all submitting 
+transactions to an emulated blockchain. Once you have defined a suitable model, 
+then QuickCheck can generate and run many thousands of scenarios, taking the 
+application through a wide variety of states, and checking that it behaves correctly 
+in each one. 
+
+   See also: :doc:`Testing Plutus Contracts with Contract Models <../tutorials/contract-models>`. 
+
```

Comment:
There is another link to the 2nd tutorial on adversarial testing we should add here. Lets really sell this one its good stuff!

Reply:
@asutherlandus Are you talking about "Property-based testing of Plutus contracts"? https://plutus-apps.readthedocs.io/en/latest/plutus/tutorials/contract-testing.html